pytest_fixtures/satellite_auth.py

```diff
@@ -145,3 +146,60 @@ def enroll_idm_and_configure_external_auth():
         f"--server {settings.ipa.hostname_ipa} "
         f"--realm {domain.upper()} -U"
     )
+
+
+@fixture(scope='session')
+def enroll_ad_and_configure_external_auth():
+    """Enroll Satellite Server to an AD Server."""
+    packages = (
+        'sssd adcli realmd ipa-python-compat krb5-workstation '
+        'samba-common-tools gssproxy nfs-utils ipa-client'
+    )
+    realm = settings.ldap.realm
+    workgroup = realm.split(".")[0]
+
+    default_content = f"[global]\nserver = unused\nrealm = {realm}"
+    keytab_content = (
+        f"[global]\nworkgroup = {workgroup}\nrealm = {realm}"
+        f"\nkerberos method = system keytab\nsecurity = ads"
+    )
+
+    # install the required packages
+    run_command(cmd=f'yum -y --disableplugin=foreman-protector install {packages}')
+
+    # update the AD name server
+    run_command(cmd="chattr -i /etc/resolv.conf")
+    result = run_command(
+        cmd="awk -v search='nameserver' '$0~search{print NR; exit}' /etc/resolv.conf"
+    )
+    line_number = int(''.join(result))
+    run_command(
+        cmd=f'sed -i "{line_number}i nameserver {settings.ldap.nameserver}" /etc/resolv.conf'
+    )
+    run_command(cmd="chattr +i /etc/resolv.conf")
+
+    # join the realm
+    run_command(cmd=f"echo {settings.ldap.password} | realm join -v {realm}")
+    run_command(cmd="touch /etc/ipa/default.conf")
```

Comment:
I think you do not need to use "touch" to create the file, echo will do it for you.

Reply:
I'm following steps from here https://access.redhat.com/documentation/en-us/red_hat_satellite/6.7/html/administering_red_hat_satellite/chap-red_hat_satellite-administering_red_hat_satellite-configuring_external_authentication#configuring-direct-ad-integration-with-gss-proxy_admin